### PR TITLE
download directory and ui bug fixes for windows

### PR DESCRIPTION
**Pull request by jackrobison**

### CODE REVIEW

File: lbrynet/reflector/client/client.py [hunk 1]
```diff
@@ -407,6 +407,10 @@ def send_blob_info(self):
             'blob_size': self.next_blob_to_send.length
         }))
 
+    def log_fail_and_disconnect(self, err, blob_hash):
+        log.error("Error reflecting blob %s", blob_hash)
+        self.transport.loseConnection()
```
Comment (jackrobison) [inline lbrynet/reflector/client/client.py]: This probably isn't ideal, not sure how to best approach. @jobevers

File: lbrynet/reflector/client/client.py [hunk 2]
```diff
@@ -420,7 +424,7 @@ def send_next_request(self):
             d = self.blob_manager.get_blob(blob_hash, True)
             d.addCallback(self.open_blob_for_reading)
             # send the server the next blob hash + length
-            d.addCallback(lambda _: self.send_blob_info())
+            d.addCallbacks(lambda _: self.send_blob_info(), lambda err: self.log_fail_and_disconnect(err, blob_hash))
```
Comment (jobevers) [inline lbrynet/reflector/client/client.py]: can you put this on two lines.  I have a hard time reading this to see what the two callbacks are?

File: lbrynet/reflector/reupload.py [hunk 3]
```diff
@@ -39,7 +39,13 @@ def _reflect_if_unavailable(reflector_has_stream, lbry_file, reflector_server):
...
 
+def _catch_error(err, uri):
+    log.error("An error occurred while checking availability for lbry://%s", uri)
+    log.debug("Traceback: %s", err.getTraceback())
+
+
 def check_and_restore_availability(lbry_file, reflector_server):
     d = _check_if_reflector_has_stream(lbry_file, reflector_server)
-    d.addCallback(lambda send_stream: _reflect_if_unavailable(send_stream, lbry_file, reflector_server))
+    d.addCallbacks(lambda send_stream: _reflect_if_unavailable(send_stream, lbry_file, reflector_server),
```
Comment (jobevers) [inline lbrynet/reflector/reupload.py]: two lines would be nice here too

File: lbrynet/reflector/reupload.py [hunk 4]
```diff
@@ -39,7 +39,13 @@ def _reflect_if_unavailable(reflector_has_stream, lbry_file, reflector_server):
     return _reflect_stream(lbry_file, reflector_server)
 
 
+def _catch_error(err, uri):
+    log.error("An error occurred while checking availability for lbry://%s", uri)
+    log.debug("Traceback: %s", err.getTraceback())
```
Comment (jobevers) [inline lbrynet/reflector/reupload.py]: I added `log_support.failure` recently: https://github.com/lbryio/lbry/blob/master/lbrynet/core/log_support.py#L152

File: lbrynet/reflector/client/client.py (hunk 1 above)
Comment (jobevers) [inline lbrynet/reflector/client/client.py]: could have sworn I replied to this - I think losing the connection is fine.  Maybe we'll want to have the caller retry at some point?

**Comment by jobevers** [comment]
I'm merging these, but would be nice if the few comments were fixed-up at some point soon.

**Pull request merged**